[quant][fx][graphmode][refactor] Change activation_post_process_map to track the observer name instead

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #54928 [quant][graphmode][fx][refactor] Quantize by Use of a Tensor instead of Tensor
* #54924 [quant][graphmode][fx] Produce torch.cat instead of torch.ops.quantized.cat
* #54813 [quant][graphmode][fx] Optimize cat
* #54812 [nn] Add allow_duplicate option for named_modules
* #54733 [quant][graphmode][fx][refactor] Factor out insert_observers_for_model to a separate function
* #54644 [quant][graphmode][fx] Separate handling Copy operator to a helper function
* **#54643 [quant][fx][graphmode][refactor] Change activation_post_process_map to track the observer name instead**
* #54073 [quant][graphmode][refactor] Remove reduandent code
* #53586 [quant][graphmode][fx] Add support for one value being quantized with different qconfigs

Summary:
A refactor needed for future changes.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27314677](https://our.internmc.facebook.com/intern/diff/D27314677)

jerryzh168 committed:
[quant][fx][graphmode][refactor] Change activation_post_process_map to track the observer name instead

Summary:
A refactor needed for future changes.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented on torch/quantization/fx/quantize.py:
```diff
@@ -377,7 +377,7 @@ def __init__(self):
         # mapping from matched node to activation_post_process
         # must be filled before convert
         self.activation_post_process_map: Optional[
```
could we also update the comment to reflect the new code?

jerryzh168 committed:
Update on "[quant][fx][graphmode][refactor] Change activation_post_process_map to track the observer name instead"


Summary:
A refactor needed for future changes.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27314677](https://our.internmc.facebook.com/intern/diff/D27314677)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fx][graphmode][refactor] Change activation_post_process_map to track the observer name instead"


Summary:
A refactor needed for future changes.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27314677](https://our.internmc.facebook.com/intern/diff/D27314677)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fx][graphmode][refactor] Change activation_post_process_map to track the observer name instead"


Summary:
A refactor needed for future changes.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27314677](https://our.internmc.facebook.com/intern/diff/D27314677)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fx][graphmode][refactor] Change activation_post_process_map to track the observer name instead"


Summary:
A refactor needed for future changes.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27314677](https://our.internmc.facebook.com/intern/diff/D27314677)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fx][graphmode][refactor] Change activation_post_process_map to track the observer name instead"


Summary:
A refactor needed for future changes.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27314677](https://our.internmc.facebook.com/intern/diff/D27314677)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fx][graphmode][refactor] Change activation_post_process_map to track the observer name instead"


Summary:
A refactor needed for future changes.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27314677](https://our.internmc.facebook.com/intern/diff/D27314677)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fx][graphmode][refactor] Change activation_post_process_map to track the observer name instead"


Summary:
A refactor needed for future changes.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27314677](https://our.internmc.facebook.com/intern/diff/D27314677)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fx][graphmode][refactor] Change activation_post_process_map to track the observer name instead"


Summary:
A refactor needed for future changes.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27314677](https://our.internmc.facebook.com/intern/diff/D27314677)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fx][graphmode][refactor] Change activation_post_process_map to track the observer name instead"


Summary:
A refactor needed for future changes.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27314677](https://our.internmc.facebook.com/intern/diff/D27314677)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fx][graphmode][refactor] Change activation_post_process_map to track the observer name instead"


Summary:
A refactor needed for future changes.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27314677](https://our.internmc.facebook.com/intern/diff/D27314677)

[ghstack-poisoned]
